_posts/2024-09-27-learning-a-programming-language-with-generative-ai.md

```diff
@@ -0,0 +1,58 @@
+---
+layout: post
+title: Learning a programming language with generative ai
+author: andandersson
+excerpt: >
+  Experiences from learning a programing language with generatve ai.
+tags:
+ - Generative ai
+ - C#
+ - .net
+---
+
+# Background
+
+Generative artificial intelligence, or GenAI, is one of the biggest hypes in the software industry. With the introduction of applications like ChatGPT, AI-algorithms can produce text, generate images and even motion pictures and music. Due to the trends, there have been many discussions how GenAI may affect software development. How can it be utilized,for example for coding, learning new tools and frameworks, code reviewing? 
+
+![A genAI-created picture of a robot coding](/img/2024-09-27-learning-a-programming-language-with-generative-ai/coding-robot.png)
+
+In the GenAI-twin project, we wanted to find out these things. For the period of March-September 2024, we started to develop a full stack application in C# with .NET and React with the help of GenAI. In that way, we were able to get a picture of how GenAI can affect software development.
+
+As a part of the journey, we also wanted to find out how GenAI can be utilized for learning a new programming language or framework. Myself as a junior developer, I did not have any prior knowledge of neither C# nor .NET and was assigned with the task.
```

Comment:
shouldn't it be "either/or" ?

was assigned "the task"? what task?

Reply:
"neither"/"nor" was suggested by a spell check, I think both are correct.

Reply:
the task of learning .NET and C# as is stated in the text.

Reply:
I agree with Tatu, that "the task" feels a bit off and should probably be specified better

Reply:
corrected